### PR TITLE
Compile flag fixes for latest clang and Z3 dependency reduction

### PR DESCRIPTION
**Pull request by katringoogoo**
- Removed Z3 dependencies from the heuristic lib
- Added the ability to build python bindings without Z3
- Fixed compile flags for clang on apple silicon

### CODE REVIEW

**Comment by burgholzer** [comment]
Thanks for working on this.

Is this PR targeted at the `apple-silicon-wheels` branch on purpose? Unfortunately this does not trigger all the CI tests, so it is hard to see if everything still works properly. I am fine with merging this and continuing in #14 if that is an option.

File: README.md [hunk 1]
```diff
@@ -72,7 +72,7 @@ JKQ QMAP is developed as a C++ library with an easy to use Python interface.
     ```
   However, in order to get the best performance out of QMAP, it is recommended to build it locally from the source distribution (see [system requirements](#system-requirements)) via
     ```bash
-    pip install --no-binary jkq.qmap
+    pip install --no-binary jkq.qmap jkq.qmap
```
Comment (burgholzer) [inline README.md:75]:
```suggestion
    pip install  jkq.qmap --no-binary jkq.qmap
```
Wouldn't it be more intuitive to write it in this fashion?

Comment (katringoogoo) [reply README.md]: right :)

**Comment by katringoogoo** [comment]
well i started the 'apple-silicon-wheels-fixes' branch from this one. i don't have anything against merging it directly to your main/master branch but i still wanted to figure out why there are so many commits showing up ...

**Comment by burgholzer** [comment]
Good question. These commits seem to originate from the commit history of a different branch... no idea how they got there...

Since #14 is on hold for now (since it's quite a hassle to get cross-compilation to work with external dependencies), I would suggest that this PR should probably target the `master` branch for now, so that it can be merged soon 👍🏼

**Comment by katringoogoo** [comment]
> Good question. These commits seem to originate from the commit history of a different branch... no idea how they got there...
> 
> Since #14 is on hold for now (since it's quite a hassle to get cross-compilation to work with external dependencies), I would suggest that this PR should probably target the `master` branch for now, so that it can be merged soon 👍🏼

I'm going to start fresh from the master branch and will close this one - changing the target doesn't really affect the commits ...

**Comment by burgholzer** [comment]
Sounds good. Sorry for the inconvenience 🙃

**Comment by katringoogoo** [comment]
ok yeah i merged in @IsFairy's branch because she fixed a z3 binding issue recently and it seems like the commits on her branch where squashed and then merged to master and now they are showing up again. i'm gonna cherry pick what i need and base it on the master branch as i said before :)

**Pull request closed without merging**